### ISSUE TITLE
Failure to process recent wikidata JSON dumps - "Unknown JSON datatype geo-shape"

### ISSUE THREAD

**Issue by RichardAllanBrown** [Wikidata/Wikidata-Toolkit#258]
When using more recent wikidata JSON dumps (20170529 or 20170619),  I get an `Unknown JSON datatype "geo-shape"` thrown that aborts processing. Version 0.7.0 used.

configured ini file contains:
```
[general]
fProp = P345,P1476,P31,P166,P166,P805,P1346,P585,P1411,P2453,P915,P17,P297,P298,P218,P570,P27,P19,P17,P21,P569,P214,P646,P2671,P2013,P2002,P161,P453,P1431,P1040,P58,P86,P57,P344,P162,P272,P840,P364,P2130

[rdf-statements]
action = rdf
rdftasks = labels,items,entities,properties,interpropertylinks,statements,simplestatements
compression = gz
output = wikidata-statements.nt
```

When run, the JSON is successfully downloaded and starts processing entities (gets to ~520k or so) and then the following exception is thrown.

```
Exception in thread "main" java.lang.IllegalArgumentException: Unknown JSON datatype "geo-shape"
        at org.wikidata.wdtk.datamodel.json.jackson.JacksonDatatypeId.getDatatypeIriFromJsonDatatype(JacksonDatatypeId.java:133)
        at org.wikidata.wdtk.datamodel.json.jackson.JacksonDatatypeId.<init>(JacksonDatatypeId.java:152)
        at org.wikidata.wdtk.datamodel.json.jackson.JacksonPropertyDocument.getDatatype(JacksonPropertyDocument.java:106)
        at org.wikidata.wdtk.datamodel.helpers.DatamodelConverter.copy(DatamodelConverter.java:498)
        at org.wikidata.wdtk.datamodel.interfaces.EntityDocumentProcessorFilter.processPropertyDocument(EntityDocumentProcessorFilter.java:66)
        at org.wikidata.wdtk.dumpfiles.JsonDumpFileProcessor.handleDocument(JsonDumpFileProcessor.java:133)
        at org.wikidata.wdtk.dumpfiles.JsonDumpFileProcessor.processDumpFileContents(JsonDumpFileProcessor.java:91)
        at org.wikidata.wdtk.dumpfiles.DumpProcessingController.processDumpFile(DumpProcessingController.java:546)
        at org.wikidata.wdtk.dumpfiles.DumpProcessingController.processDump(DumpProcessingController.java:482)
        at org.wikidata.wdtk.client.Client.performActions(Client.java:170)
        at org.wikidata.wdtk.client.Client.main(Client.java:295)
java.io.IOException: Write end dead
        at java.io.PipedInputStream.read(PipedInputStream.java:311)
        at java.io.PipedInputStream.read(PipedInputStream.java:378)
        at java.io.InputStream.read(InputStream.java:101)
        at org.wikidata.wdtk.client.DumpProcessingOutputAction$1.run(DumpProcessingOutputAction.java:316)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by Tpt** [comment]
This branch contains a fix for this issue (and will avoid failures when Wikidata introduces new datatypes): https://github.com/Tpt/Wikidata-Toolkit/tree/datatype-normalization

**Comment by Tpt** [comment]
Fixed by change #260. WikidataToolkit should not fail anymore when new datatypes are introduced.
